test/e2e/pods.go

```diff
@@ -30,10 +30,59 @@ import (
 	. "github.com/onsi/gomega"
 )
 
+func runLivenessTest(c *client.Client, podDescr *api.Pod) {
+	defer GinkgoRecover()
+	ns := "e2e-test-" + string(util.NewUUID())
+
+	By(fmt.Sprintf("Creating pod %s in namespace %s", podDescr.Name, ns))
+	_, err := c.Pods(ns).Create(podDescr)
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("creating pod %s", podDescr.Name))
+
+	// At the end of the test, clean up by removing the pod.
+	defer func() {
```

Comment:
By() takes an optional second argument that is a callback function. So this can be written as:

defer By("deleting the pod", c.Pods(ns).Delete(podDescr.Name))